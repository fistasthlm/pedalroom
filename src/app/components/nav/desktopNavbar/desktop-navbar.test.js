import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';
import DesktopNavbar from './desktop-navbar';

const setup = propOverrides => {
    const props = Object.assign({
        links: List(),
    }, propOverrides);

    const wrapper = shallow(<DesktopNavbar {...props} />);

    return {
        setup,
        wrapper
    };
};

describe('Render', () => {
    it('should render component', () => {
        const { wrapper } = setup();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render links', () => {
        const { wrapper } = setup({
            links: List([
                Map({
                    to: '/',
                    route: 'Home',
                }),
                Map({
                    to: '/bikes',
                    route: 'Bikes',
                }),
            ]),
        });

        expect(wrapper).toMatchSnapshot();
    });
});
