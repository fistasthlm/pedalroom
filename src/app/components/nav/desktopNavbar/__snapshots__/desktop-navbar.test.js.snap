// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="navbar"
/>
`;

exports[`Render should render links 1`] = `
<div
  className="navbar"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="nav-item"
    exact={true}
    key="0-/"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="nav-item"
    exact={true}
    key="1-/bikes"
    to="/bikes"
  >
    Bikes
  </NavLink>
</div>
`;
